packages/broker/src/plugins/operator/inspectionUtils.ts

```diff
@@ -0,0 +1,183 @@
+        logger.info('Skip inspection (no suitable operator)', { targetSponsorship })
+        return undefined
+    }
+
+    // choose stream part
+    const streamParts = getPartitionsOfStreamAssignedToMe(loadBalancer, targetSponsorship.streamId)
+    const targetStreamPart = sample(streamParts)
+    if (targetStreamPart === undefined) {
+        // Only happens if during the async awaits the stream parts I am assigned to have changed.
+        logger.info('Skip inspection (no suitable stream part)', { targetSponsorship, targetOperatorAddress })
+        return undefined
+    }
+
+    return {
+        sponsorshipAddress: targetSponsorship.sponsorshipAddress,
+        operatorAddress: targetOperatorAddress,
+        streamPart: targetStreamPart
+    }
+}
+
+export async function findNodesForTarget(
+    target: Target,
+    streamrClient: StreamrClient,
+    fetchRedundancyFactorFn: FetchRedundancyFactorFn,
+    maxWait: number,
+    abortSignal: AbortSignal
+): Promise<NetworkPeerDescriptor[]> {
+    logger.debug('Waiting for node heartbeats', {
+        targetOperator: target.operatorAddress,
+        maxWait
+    })
+    const targetOperatorFleetState = new OperatorFleetState(
+        streamrClient,
+        toStreamID('/operator/coordination', target.operatorAddress)
+    )
+    try {
+        await targetOperatorFleetState.start()
+        await Promise.race([
+            targetOperatorFleetState.waitUntilReady(),
+            wait(maxWait, abortSignal)
+        ])
+        logger.debug('Finished waiting for heartbeats', {
+            targetOperator: target.operatorAddress,
+            onlineNodes: targetOperatorFleetState.getNodeIds().length,
+        })
+
+        const replicationFactor = await fetchRedundancyFactorFn({
+            operatorContractAddress: target.operatorAddress,
+            signer: await streamrClient.getSigner()
+        })
+        if (replicationFactor === undefined) {
+            logger.debug('Encountered misconfigured replication factor')
+            return []
+        }
+
+        const consistentHashRing = new ConsistentHashRing(replicationFactor)
+        for (const nodeId of targetOperatorFleetState.getNodeIds()) {
+            consistentHashRing.add(nodeId)
+        }
+        const targetNodes = consistentHashRing.get(target.streamPart)
+        return targetNodes.map((nodeId) => targetOperatorFleetState.getPeerDescriptor(nodeId)!)
+    } finally {
+        await targetOperatorFleetState.destroy()
+    }
+}
+
+export async function inspectTarget({
+    target,
+    streamrClient,
+    fetchRedundancyFactor,
+    heartbeatLastResortTimeoutInMs,
+    abortSignal,
+    findNodesForTargetFn = findNodesForTarget
+}: {
+    target: Target
+    streamrClient: StreamrClient
+    fetchRedundancyFactor: FetchRedundancyFactorFn
+    heartbeatLastResortTimeoutInMs: number
+    abortSignal: AbortSignal
+    findNodesForTargetFn?: typeof findNodesForTarget
```

Comment:
Maybe this would simpler if we'd pass the `targetPeerDescriptors` as a parameter instead of `findNodesForTargetFn`? This function just calls the `findNodesForTargetFn` once.